main/src/main/java/org/mobilitydata/gtfsvalidator/util/HtmlOutputUtil.java

```diff
@@ -0,0 +1,199 @@
+  private static final String WARNING = SeverityLevel.WARNING.name();
+  private static final String INFO = SeverityLevel.INFO.name();
+  private static final String NOTICES = "notices";
+  private static final String CODE = "code";
+  private static final String SEVERITY = "severity";
+  private static final String TOTAL_NOTICES = "totalNotices";
+  private static final String SAMPLE_NOTICES = "sampleNotices";
+
+  public static String outputBuilder(
+      JsonObject noticesJson, String jqueryDependency, String styleCss, String scriptJs) {
+    StringBuilder notices = new StringBuilder();
+    HashMap<String, Integer> noticesQty =
+        new HashMap<>() {
+          {
+            put(ERROR, 0);
+            put(WARNING, 0);
+            put(INFO, 0);
+          }
+        };
+    JsonArray noticesArray = noticesJson.getAsJsonArray(NOTICES);
+    for (JsonElement notice : noticesArray) {
+      JsonObject noticeJson = notice.getAsJsonObject();
+      String noticeCode = noticeJson.get(CODE).getAsString();
+      String noticeSeverity = noticeJson.get(SEVERITY).getAsString();
+      int totalNotices = noticeJson.get(TOTAL_NOTICES).getAsInt();
+      noticesQty.put(noticeSeverity, noticesQty.get(noticeSeverity) + totalNotices);
+      JsonArray sampleNotices = noticeJson.getAsJsonArray(SAMPLE_NOTICES);
+      JsonElement sampleNotice = sampleNotices.get(sampleNotices.size() - 1);
+      JsonObject sampleJson = sampleNotice.getAsJsonObject();
+      ArrayList<String> noticeFields = new ArrayList<>();
+      Set<Map.Entry<String, JsonElement>> sampleEntries = sampleJson.entrySet();
+      for (Map.Entry<String, JsonElement> entry : sampleEntries) {
+        noticeFields.add(entry.getKey());
+      }
+      String sampleColumns = noticeColumnsBuilder(noticeFields);
+      String sampleRows = noticeRowsBuilder(noticeFields, sampleNotices);
+      notices.append(
+          noticeFrame(noticeCode, noticeSeverity, totalNotices, sampleColumns, sampleRows));
+    }
+    String outputHeader =
+        outputHeader(
+            (noticesQty.get(ERROR) + noticesQty.get(WARNING) + noticesQty.get(INFO)),
+            noticesQty.get(ERROR),
+            noticesQty.get(WARNING),
+            noticesQty.get(INFO));
+    return outputFrame(
+        jqueryDependency, styleCss, scriptJs, outputHeader, notices.toString(), outputFooter());
+  }
+
+  public static String resourceLoader(String resourcePath) {
+    String content = "";
+    try (InputStream stream =
+        HtmlOutputUtil.class.getClassLoader().getResourceAsStream(resourcePath)) {
+      Scanner scanner = new Scanner(stream).useDelimiter("\\A");
+      content = scanner.hasNext() ? scanner.next() : "";
+    } catch (IOException e) {
+      System.out.println("The resource was not found at \"" + resourcePath + "\".");
+    }
+    return content;
+  }
+
+  protected static String noticeColumnsBuilder(ArrayList<String> noticeFields) {
+    StringBuilder columns = new StringBuilder("                ");
```

Comment:
The current HTML representation is definitely an improvement over digging through raw JSON, but I think the workflow for the HTML report could be improved further. Currently it's still very difficult to track down what an error actually means.

Right now, we show the user the list of errors/warnings:

![image](https://user-images.githubusercontent.com/928045/166062229-3e128f16-03df-4390-b11a-038d27cc97c0.png)

The natural question is "what does this error mean?"

So I would click on NOTICES.md and RULES.md and search for the code like `equal_shape_distance_diff_coordinates` to find more info.

However, if I search RULES.md, I don't get any matches because the rules are listed by notice class name and not the error code.

If I search NOTICES.md, I do get a hit in a list of notice codes mapped to notice class names:

![image](https://user-images.githubusercontent.com/928045/166062592-f2fa98b6-66c9-4394-aef8-6fb75dbdb7eb.png)

...but that's not very helpful by itself. If I click on the notice class name I'm taken to an anchor lower on the same NOTICES.md page that describes what the fields within that notice mean:

https://github.com/MobilityData/gtfs-validator/blob/master/docs/NOTICES.md#EqualShapeDistanceDiffCoordinatesNotice

![image](https://user-images.githubusercontent.com/928045/166062939-4b6c1a30-6cc5-4ce2-b593-334d36b3ca9b.png)

But that still doesn't describe what the actual error means. If I click on the notice class name linked in this anchor THEN I get taken over to RULES.md for the description of the error that I was originally looking for:

https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#EqualShapeDistanceDiffCoordinatesNotice

![image](https://user-images.githubusercontent.com/928045/166063143-4a1fa9ce-47b2-4c7c-bd7d-2c4e0eb8302d.png)

IMHO in the HTML report itself we should link directly to these error descriptions so the user has a single click to find out what the error means. We did this in the GTFS Realtime Validator HTML so the user is directly linked to, for example:
https://github.com/MobilityData/gtfs-realtime-validator/blob/master/RULES.md#e050---timestamp-is-in-the-future

...when clicking on that error in the report.

Maybe a simple approach is just to add the error code to RULES.md for now, maybe as section anchors in the markdown if that's possible so you can link directly to them?

That's what we did with RT validator - see https://raw.githubusercontent.com/MobilityData/gtfs-realtime-validator/master/RULES.md and anchors like `<a name="E003"/>`.

Likely beyond the scope of this issue, but I think we should also move the field descriptions in NOTICES.md into RULES.md, possibly as [collapsed text](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) that you can expand if you want the field details.

Reply:
@barbeau this is a great piece of feed-back. 
We were thinking of having this PR merged as a first step and then implementing (at least) the two additional parts you flagged here: 
1. A description of the notice (what is this problem?)
2. Where exactly is the issue and how to fix it (How to fix my dataset?)

Before implementing these parts, we would like to (1) gather user feedback & do some user testing on different options for the next version of the HTML output, (2) discuss the best strategy to integrate the info from RULES.md and NOTICES.md into the results (knowing that we will want to translate).

> Maybe a simple approach is just to add the error code to RULES.md for now, maybe as section anchors in the markdown if that's possible so you can link directly to them?
I really like this idea and this would be actionable for the next release, and I'm willing to PR this one. What about merging RULES.md and NOTICES.md in one document, using the collapsed text to make it more digestible?

Reply:
>What about merging RULES.md and NOTICES.md in one document, using the collapsed text to make it more digestible?

👍

Reply:
✅ Done in PR #1132, and the anchor links have been added in the HTML output